### PR TITLE
require trust for maven wrapper per workspace

### PR DESCRIPTION
**Pull request by Eskibear**
a security patch for #525 
image

### CODE REVIEW

File: src/utils/contextUtils.ts [hunk 1]
```diff
@@ -83,3 +83,18 @@ export function getPathToWorkspaceStorage(...args: string[]): string | undefined
...
+
+export async function trustWrapper(mvnw: string): Promise<boolean> {
+    const key: string = "trustMavenWrapper";
+    const trust: boolean | undefined = EXTENSION_CONTEXT.workspaceState.get<boolean | undefined>(key);
+    if (trust === undefined) {
+        const msg: string = `Maven Wrapper is found in this workspace, do you trust it? (${mvnw})`;
+        const YES: string = "Yes";
+        const NO: string = "No";
+        const choice: string | undefined = await window.showInformationMessage(msg, YES, NO);
+        EXTENSION_CONTEXT.workspaceState.update(key, choice === YES);
```
Comment (testforstephen) [inline src/utils/contextUtils.ts:95]: What happened when i click `esc`?

Comment (Eskibear) [reply src/utils/contextUtils.ts:95]: same as `No`.  it fallbacks to search for `mvn` installed locally.

Comment (Eskibear) [reply src/utils/contextUtils.ts:95]: Do you want me to keep poping it up on next time, if the user press `esc` ?

**Comment by jahan01** [comment]
perhaps a command to clear the trust configuration would help if mistakenly pressed "No"

**Comment by Eskibear** [comment]
> perhaps a command to clear the trust configuration would help if mistakenly pressed "No"

@jahan01  Now, whenever value of  `maven.executable.preferWrapper` changes (indicating you want to turn on/off using maven wrapper), it would clear the trust configuration.  I think it somehow unblock the case you mentioned.

**Review by jdneo — APPROVED** [review]
LGTM

**Pull request merged**